### DISCUSSION TITLE
Explicit outputs about battery ageing

### DISCUSSION THREAD

**Discussion by OscarPindaro** [pybamm-team/PyBaMM#1474] [Q&A] [answered]
Hello!
I'm working for a thesis on a battery controller that aims to manage a battery system while reducing the effects of ageing fenomena. The controller should be able to work properly with a very broad range of battery.
I need a framework that is able to simulate the effect of the control on the battery in order to build a dataset, since we are using machine learning techniques.
In particular, i'm interested in the capacity fade and the resistance growth ageing, and in the modeling of the State of Health (SOH).
Is there a way in your library to easily access this info? For example, i would like some like resultSimulation.capacityLoss, or resultSimulation.currentImpedence.
I don't really have a Chemistry background, so I don't know if there are some formulas that extract this from some more "lower level" outputs, and searching them would be very time consuming.
Thank you so much for yor time!

**Comment by brosaplanella** [comment]
Hi Oscar! You can track the discharge capacity after solving your simulation `sim` by calling `sim.solution["Discharge capacity [A.h]"] and the internal resistance by calling `sim.solution["Local ECM resistance [Ohm]"]. For a full list of all the accessible variables you can check [this notebook](https://colab.research.google.com/github/pybamm-team/PyBaMM/blob/main/examples/notebooks/Getting%20Started/Tutorial%203%20-%20Basic%20plotting.ipynb).

**Comment by valentinsulzer** [comment, marked as answer]
Once #1459 is merged you will be able to access `solution.summary_variables` which has lots of information about different measures of SOH as a function of cycle number
